### ISSUE TITLE
Can we get a populate function

### ISSUE THREAD

**Issue by talha-asad** [jagi/meteor-astronomy#235]
I really liked the mongoose ```populate``` feature for fetching related documents from another collection. Do you think it's possible for us to have a similar feature in v2.0?

Or perhaps there is already an easy way to do this which i am not familiar with.

**Comment by jagi** [comment]
Yes it's quite nice feature. I was thinking about similar feature that would be integrated into relations module. For now there is no such feature in Astronomy. Probably I will add such feature into relations module or integrate it into the core in 2.1 or something like this. I would definitely want to have something similar in Astronomy 2.0

**Comment by talha-asad** [comment]
Yes having a simple helper like that in relations module would be really handy.

I am guessing relations module is going into the core starting 2.0?

**Comment by jagi** [comment]
I wanted it to be in the core of 2.0 but we will see. I have many things right now on my head (You know private stuff). I wanted to release RC version before end of the year, but it won't happen so fast. It will be probably beginning of the 2016. I would like to include in this version as many features as possible.

**Comment by lsunsi** [comment]
That'd be awesome.
I was actually kind of waiting the 2.0 launch to try and code something like this.

**Comment by talha-asad** [comment]
Do you guys have a common pattern you use for this at the moment?

**Comment by jagi** [comment]
You just have to get related records using methods. You can create method like for example `getAuthor()` that will take the `authorId` field's value and fetch document from the `Meteor.users` collection and put it into the original document under `author` field.

```js
Doc = Astro.Class({
  name: 'Doc',
  collection: Docs,
  fields: {
    authorId: 'string',
    author: 'null' /* or 'object' */
  },
  methods: {
    getAuthor: function() {
      this.author = Meteor.users.findOne(this.authorId);
    }
  }
});

var doc = Docs.findOne();
doc.getAuthor(); // equivalent to 'doc.populate("author");
doc.author;
```

**Comment by talha-asad** [comment]
Thanks for the help @jagi. Appreciate it.

**Comment by ghost** [comment]
+1

**Comment by skynightz93** [comment]
Hi, do this feature already in v2?

**Comment by lukejagodzinski** [comment]
@nizulzaim no this feature is not yet implemented
